lib/valueflow.cpp

```diff
@@ -778,6 +778,38 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
         }
     }
 
+    // increment
+    else if (parent->str() == "++") {
+        for (const ValueFlow::Value &val : tok->values()) {
+            if (!val.isIntValue() && !val.isFloatValue())
+                continue;
+            ValueFlow::Value v(val);
+            if (parent == tok->previous()) {
+                if (v.isIntValue())
+                    v.intvalue = v.intvalue + 1;
+                else
+                    v.floatValue = v.floatValue + 1.0;
```

Comment:
I guess it does not hurt.. but I have the feeling that increments can't be used on floats. Maybe convert that value to an int..

Reply:
Increment operator can be used on floats. It can be used on any non-boolean type according to [here](https://en.cppreference.com/w/cpp/language/operator_incdec):

> For non-boolean operands, the expression ++x is exactly equivalent to x += 1, and the expression --x is exactly equivalent to x -= 1, that is, the prefix increment or decrement is an lvalue expression that identifies the modified operand. All arithmetic conversion rules and pointer arithmetic rules defined for arithmetic operators apply and determine the implicit conversion (if any) applied to the operand as well as the return type of the expression.

Also, converting to an int would produce an incorrect result as well. Incrementing `0.5` should result in `1.5` which is not possible if its converted to an int.

Reply:
ok thanks I did not know this :-)